libia2/padding.ld

```diff
@@ -10,21 +10,19 @@
    __start_$SECTION and before __stop_$SECTION.
 */
 SECTIONS {
-    ia2_shared_rodata : {
-        . = ALIGN(4096);
+    ia2_shared_rodata ALIGN(4096): {
         __start_ia2_shared_rodata = .;
         *(ia2_shared_rodata)
-        . = ALIGN(4096);
+        . = ALIGN(. != 0 ? 4096 : 1);
```

Comment:
I think it's just to ensure the start and stop symbols are at the same address so we don't load a page unnecessarily if ia2_shared_rodata is empty.

Reply:
I think lld doesn't support `A ? B : C` in some contexts, but the change itself seems fine to me.

Reply:
I took this from the ld default linker script. I'll try without it, but I think we will end up with a page if we don't do this.

Reply:
Yes, I tested this, it's required to avoid emitting an empty section. We don't need this section to exist if its size is 0, and this is what is required to do that...